pkg/authorization/authorizer/adapter/attributes.go

```diff
@@ -70,6 +76,26 @@ func (a AdapterAttributes) GetNamespace() string {
 	return a.namespace
 }
 
+func (a AdapterAttributes) GetName() string {
+	return a.AuthorizationAttributes.GetResourceName()
+}
+
+func (a AdapterAttributes) GetSubresource() string {
+	tokens := strings.SplitN(a.AuthorizationAttributes.GetResource(), "/", 2)
+	if len(tokens) != 2 {
+		return ""
+	}
+	return tokens[1]
+}
+
+func (a AdapterAttributes) GetResource() string {
+	tokens := strings.SplitN(a.AuthorizationAttributes.GetResource(), "/", 2)
```

Comment:
this seems like it propagates the bug... `AdapterAttributes` satisfying `oauthorizer.AuthorizationAttributes` with a `GetResource` impl that does not include subresource is scary

we should update `oauthorizer.AuthorizationAttributes` to also have a GetSubresource(), and update `node_auth.go` to set resource and subresource correctly